sklearn/tests/test_pipeline.py

```diff
@@ -1526,6 +1526,48 @@ def fit(self, X, y):
     check_is_fitted(pipeline)
 
 
+def test_feature_union_check_if_fitted():
+    class Estimator(BaseEstimator):
+        def fit(self, X, y):
+            self.fitted_ = True
+            return self
+
+        def transform(self, X):
+            return X
+
+    union = FeatureUnion([("clf", Estimator())])
+    with pytest.raises(NotFittedError):
+        check_is_fitted(union)
+
+    union.fit(iris.data, iris.target)
+    check_is_fitted(union)
+
+    # passthrough is stateless
+    union = FeatureUnion([("pass", "passthrough")])
+    check_is_fitted(union)
+
+    union = FeatureUnion([("clf", Estimator()), ("pass", "passthrough")])
+    with pytest.raises(NotFittedError):
+        check_is_fitted(union)
+
+    union.fit(iris.data, iris.target)
+    check_is_fitted(union)
+
+    union = clone(union)
+    with pytest.raises(NotFittedError):
+        check_is_fitted(union)
+
+    union.set_params(clf="drop")
+    check_is_fitted(union)
+
+    union.set_params(clf=Estimator())
+    with pytest.raises(NotFittedError):
+        check_is_fitted(union)
+
+    union.fit(iris.data, iris.target)
+    check_is_fitted(union)
```

Comment:
I think we can remove these tests. It's fundamentally testing that `clone` works. In other words, if `clone` works and the tests above pass, then these should pass as well.